### ISSUE TITLE
.NET Core 3.0 apps failed to run with static files in folders published as a single file

### ISSUE THREAD

**Issue by xiaoyuvax** [dotnet/core#3640]
### Describe the bug
Static files included in project but in FOLDER structure failed to run after published as a single file, eg. you Included a wwwroot in your project and set all individual files to be copied to output folders under VS which are assumed to be packed into the single executable.

running the output executable shows following messages:
> Failure processing application bundle; possible file corruption.
> Invalid FileEntry detected.
> A fatal error was encountered. Could not extract contents of the bundle
These messages were gone when abovementioned Folders were removed.
Static files in project root doesn't yield this problem.

### To Reproduce
1.Inlucde folders of files in project
2.Publish with PublishSingleFile=true 
3.Run the output executable,you get the message.

### Expected behavior
Launch as usual since the packed folders structure should be extracted to the temporary path properly.

### Relevant Issue:
[https://github.com/aspnet/AspNetCore-Tooling/pull/820](url)
[https://github.com/aspnet/AspNetCore/issues/12020](url)

@DamianEdwards @davidfowl @glennc @anurse

**Comment by davidfowl** [comment]
Why did you move it from ASP.NET Core?

**Comment by xiaoyuvax** [comment]
I encountered this problem during publishing a console program which deosn't depend on asp.net core at all, though i did get some information on similar situation from some asp.net threads, i think it is more proper to discuss it here.

**Comment by leecow** [comment]
@nguerrera @dsplaisted @elinor-fung

**Comment by dsplaisted** [comment]
Is single-file publish supposed to work for ASP.NET apps with content files, etc?

**Comment by elinor-fung** [comment]
@swaroop-sridhar

Single-file publish should still work properly for subfolders with subfolders.

I wasn't able to repro with:
1. Create a console app with a folder structure like so:
    ```
    │   helloWorld.csproj
    │   Program.cs
    └───subfolder
        └───sub
                test1.txt
                test2.txt
    ```
2. Set `<Content Include="subfolder\sub\**" CopyToOutputDirectory="Always"/>` in the .csproj.
3. `dotnet.exe publish /p:Configuration=Release /p:PublishSingleFile=true /p:RuntimeIdentifier=win10-x64`
4. `bin\Release\netcoreapp3.0\win10-x64\publish\helloWorld.exe`

@xiaoyuvax, did I miss a step for the console app repro? Is this with the official 3.0 release?

**Comment by swaroop-sridhar** [comment]
Yes single-file publish is expected to package all published files (except for some special files like PDBs) including sub-directory structure into the app. 

* If you publish without the `PublishSingleFile` option, are the sub-folders in the publish directory? 
* If so, when you run the app, please check the extraction location (typically `<tmp-directory>/.net/<app-name>/<some-hash>`) to see if the content files were extracted.

Thanks.

**Comment by xiaoyuvax** [comment]
First of all, I did more trials by setting off all extra configurations such as TierCompilation and PublishTrimmed and reproduced the same result:

> Failure processing application bundle; possible file corruption.
> Invalid FileEntry detected.
> A fatal error was encountered. Could not extract contents of the bundle
As i've described in the first post.

@swaroop-sridhar 

- Without PublishSingleFile option, subfolders are there in the publish folder
- Once run after PublishedSingle-File with message above shown, only see a four-digits random numbered folder created in the temp folder without any contents inside, nothing.

@elinor-fung 
I created a new console project with following config. still got the same problem.

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <PublishSingleFile>true</PublishSingleFile>
    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
    <ApplicationIcon />
    <StartupObject />
  </PropertyGroup>

  <ItemGroup>
    <None Remove="wwwroot\css\app.css" />
    <None Remove="wwwroot\css\chunk-0bcfab7e.css" />
    <None Remove="wwwroot\css\chunk-4a31f3fb.css" />
    <None Remove="wwwroot\css\chunk-56450eb9.css" />
    <None Remove="wwwroot\css\chunk-5b4909b2.css" />
    <None Remove="wwwroot\css\chunk-8d0177ea.css" />
    <None Remove="wwwroot\css\chunk-a456f024.css" />
    <None Remove="wwwroot\css\chunk-ad0b3224.css" />
    <None Remove="wwwroot\css\chunk-edce5902.css" />
    <None Remove="wwwroot\favicon.ico" />
    <None Remove="wwwroot\index.html" />
  </ItemGroup>

  <ItemGroup>
    <Content Include="wwwroot\css\app.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-0bcfab7e.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-4a31f3fb.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-56450eb9.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-5b4909b2.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-8d0177ea.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-a456f024.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-ad0b3224.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\css\chunk-edce5902.css">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\favicon.ico">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
    <Content Include="wwwroot\index.html">
      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
    </Content>
  </ItemGroup>

</Project>
```

**Comment by xiaoyuvax** [comment]
@dsplaisted Please read the relevant issues attached concerning contents folders in  asp.net core.

**Comment by swaroop-sridhar** [comment]
@xiaoyuvax https://github.com/aspnet/AspNetCore/issues/12020 is a different (understood) issue -- in that the content files end up in the extraction directory (rather than beside the host) when bundled into a single file.

In this case, looks like the bundle extraction failed -- I'll try to repro the failure now, and look into it.
